plugins/modules/vultr_block_storage.py

```diff
@@ -216,28 +242,140 @@ def absent_block_storage_volume(self):
                 )
         return volume
 
+    def detached_block_storage_volume(self):
+        volume = self.present_block_storage_volume()
+        if volume.get('attached_to_SUBID') is None:
+            return volume
+
+        self.result['changed'] = True
+
+        if not self.module.check_mode:
+            data = {
+                'SUBID': volume['SUBID'],
+                'live': self.module.params.get('live_attachment')
+            }
+            self.api_query(
+                path='/v1/block/detach',
+                method='POST',
+                data=data
+            )
+
+            volume = self.get_block_storage_volumes()
+        else:
+            volume['attached_to_SUBID'] = None
+
+        self.result['diff']['after'] = volume
+
+        return volume
+
+    def attached_block_storage_volume(self):
+        expected_server = self.module.params.get('attached_to_SUBID')
+        volume = self.present_block_storage_volume()
+        server = volume.get('attached_to_SUBID')
+        if server == expected_server:
+            return volume
+
+        if server is not None:
+            raise VultrException(
```

Comment:
throwing exceptions to users is considered to be a bug in ansible terms. We need to catch it and use 
```
self.module.fail_json(msg="")
```
to send error messages to the user.

Reply:
fixed, and adapted tests to account for this